src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/mha_single_token.cpp

```diff
@@ -822,7 +825,7 @@ static ov::float16 dot_product_fp16(ov::float16* a,
     float16_t sum_2 = svaddv_f16(pg, sum2);
     float16_t sum_3 = svaddv_f16(pg, sum3);
     sum = static_cast<float>(sum_0 + sum_1 + sum_2 + sum_3);
-#    else
+#    elif defined(HAVE_NEON_FP16)
```

Comment:
I checked this on macOS, and the define is not enabled.
Perhaps we can use `defined(__ARM_NEON) && defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)` instead for macOS.